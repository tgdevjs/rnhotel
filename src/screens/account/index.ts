export * from './Account';
export * from './AccountGuest';
export * from './AccountMenu';
export * from './JoinModal';
export * from './SignInModal';
