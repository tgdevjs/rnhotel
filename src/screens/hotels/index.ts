export * from './Hotels';
export * from './ConfirmReservation';
