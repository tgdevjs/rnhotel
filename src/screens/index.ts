export * from './account';
export * from './hotels';
export * from './search';
export * from './stays';
