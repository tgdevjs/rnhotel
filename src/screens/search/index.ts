export * from './Search';
export * from './SearchCalendar';
