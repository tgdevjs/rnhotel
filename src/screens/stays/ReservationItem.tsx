import React from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import Swipeout from "react-native-swipeout";
import moment from "moment";

import { DateRange, HotelTitle, DateRangeError } from "../../components";
import { ReservationType } from "../../types";
import { borderRadius, colors, fonts, spacing } from "../../styles";

type Props = {
  item: ReservationType;
  onDeleteItem: (id: string) => void;
};

const styles = StyleSheet.create({
  header: {
    alignItems: "center",
    borderColor: colors.gray,
    borderWidth: 1,
    borderRadius: borderRadius.large,
    justifyContent: "center",
    marginHorizontal: spacing,
    height: 160
  },
  textContainer: {
    paddingLeft: 42
  },
  dateRange: {
    paddingBottom: spacing
  },
  dateText: fonts.H2,
  hotelTitle: {
    color: colors.black
  },
  name: fonts.Body,
  date: {
    flexDirection: "row"
  }
});

export const ReservationItem = ({
  item: { arrivalDate, departureDate, hotelName, id },
  onDeleteItem
}: Props) => {
  return (
    <Swipeout
      right={[
        {
          text: "Delete",
          backgroundColor: colors.red,
          onPress: () => onDeleteItem(id)
        }
      ]}
      autoClose
      backgroundColor="transparent"
    >
      <TouchableOpacity onPress={() => {}}>
        <View style={styles.header}>
          {!moment(arrivalDate).isValid() ||
          !moment(departureDate).isValid() ? (
            <DateRangeError startDay={arrivalDate} endDay={departureDate} />
          ) : (
            <DateRange
              containerStyle={styles.dateRange}
              dateTextStyle={styles.dateText}
              startDay={arrivalDate}
              endDay={departureDate}
            />
          )}
          <HotelTitle style={styles.hotelTitle} title={hotelName} />
          <Text>{id}</Text>
        </View>
      </TouchableOpacity>
    </Swipeout>
  );
};
