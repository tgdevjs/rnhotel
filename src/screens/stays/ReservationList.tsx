import React from "react";
import { Query, Mutation } from "react-apollo";
import { FlatList, StyleSheet, Text, View } from "react-native";

import { ReservationItem } from ".";
import { ReservationType } from "../../types";
import {
  reservationsQuery,
  deleteReservationMutation
} from "../../apollo/queries";
import { getReservationsQueryVariables } from "../../utils";
import { fonts } from "../../styles";

type Props = {
  name: string;
  orderBy: string;
  timeRangeKey: string;
};

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  itemSeparator: {
    height: 8
  },
  listEmpty: {
    alignItems: "center",
    flex: 1,
    height: 400,
    justifyContent: "center"
  },
  actionText: fonts.H3
});

export const ReservationList = ({ name, orderBy, timeRangeKey }: Props) => {
  const variables = getReservationsQueryVariables({
    name,
    orderBy,
    timeRangeKey
  });
  const timeRange = timeRangeKey === "arrivalDate_gte" ? "upcoming" : "past";

  return (
    <Mutation
      mutation={deleteReservationMutation}
      update={(cache, { data: { deleteReservation } }) => {
        const { reservations } = cache.readQuery({
          query: reservationsQuery,
          variables
        });
        cache.writeQuery({
          query: reservationsQuery,
          variables,
          data: {
            reservations: reservations.filter(
              (item: ReservationType) => item.id !== deleteReservation.id
            )
          }
        });
      }}
    >
      {deleteReservation => (
        <Query query={reservationsQuery} variables={variables}>
          {({ data, loading }) => {
            if (loading) return null;

            return (
              <View style={styles.container}>
                <FlatList
                  data={data.reservations}
                  horizontal={false}
                  ItemSeparatorComponent={() => (
                    <View style={styles.itemSeparator} />
                  )}
                  keyExtractor={(item: ReservationType) => item.id}
                  ListEmptyComponent={
                    <View style={styles.listEmpty}>
                      <Text style={styles.actionText}>
                        {`No ${timeRange} stays`}
                      </Text>
                    </View>
                  }
                  renderItem={({ item }) => (
                    <ReservationItem
                      item={item}
                      onDeleteItem={id =>
                        deleteReservation({ variables: { where: { id } } })
                      }
                    />
                  )}
                />
              </View>
            );
          }}
        </Query>
      )}
    </Mutation>
  );
};
