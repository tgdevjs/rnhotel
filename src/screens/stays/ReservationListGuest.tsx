import React from "react";
import { StyleSheet, Text, View } from "react-native";

import { SignInButton } from "../../components";
import { fonts } from "../../styles";

type Props = {
  onSignIn: () => void;
};

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  main: {
    alignItems: "center"
  },
  actionSection: {
    margin: 20,
    alignItems: "center"
  },
  actionText: fonts.H3,
  valueText: fonts.H3
});

export const ReservationListGuest = ({ onSignIn }: Props) => {
  return (
    <View style={styles.mainContainer}>
      <View style={styles.main}>
        <View style={styles.actionSection}>
          <Text style={styles.actionText}>Sign in</Text>
          <Text style={styles.valueText}>to see your reservations</Text>
        </View>
        <SignInButton onPress={onSignIn} />
      </View>
    </View>
  );
};
