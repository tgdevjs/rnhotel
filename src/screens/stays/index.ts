export * from './Stays';
export * from './StayItem';
export * from './StaysList';
export * from './StaysListGuest';
