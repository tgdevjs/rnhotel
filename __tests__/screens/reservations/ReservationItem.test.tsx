import React from "react";
import { ReservationItem } from "../../../src/screens/stays";

describe("ReservationItem", () => {
  it("renders default", () => {
    expect(
      shallow(
        <ReservationItem
          item={{
            arrivalDate: "2019-06-01",
            departureDate: "2019-06-03",
            hotelName: "hotelName",
            id: "1"
          }}
        />
      )
    ).toMatchSnapshot();
  });
});
