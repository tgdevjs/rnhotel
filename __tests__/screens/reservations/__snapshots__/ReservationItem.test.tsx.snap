// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationItem renders default 1`] = `
<Swipeout
  autoClose={true}
  backgroundColor="transparent"
  disabled={false}
  right={
    Array [
      Object {
        "backgroundColor": "#FF0000",
        "onPress": [Function],
        "text": "Delete",
      },
    ]
  }
  rowID={-1}
  sectionID={-1}
  sensitivity={50}
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#C3C3C3",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 160,
          "justifyContent": "center",
          "marginHorizontal": 10,
        }
      }
    >
      <DateRange
        containerStyle={
          Object {
            "paddingBottom": 10,
          }
        }
        dateTextStyle={
          Object {
            "fontSize": 36,
            "fontWeight": "bold",
          }
        }
        endDay="2019-06-03"
        startDay="2019-06-01"
      />
      <HotelTitle
        style={
          Object {
            "color": "#000",
          }
        }
        title="hotelName"
      />
      <Text>
        1
      </Text>
    </View>
  </TouchableOpacity>
</Swipeout>
`;
